### ISSUE TITLE
Module Generation

### ISSUE THREAD

**Issue by sercanov** [futurice/pepperoni-app-kit#184]
It would be great to have a yeoman style module generation. Like `pepperoni generate module:Login` would create a folder in modules with defaults. Isn't it ?

**Comment by krivachy** [comment]
Yes, that's quite a nice idea. If you have time you could prototype something and we can see how it would work!

**Comment by sercanov** [comment]
Sure ! I think yeoman is not the only option here, will come up in a few days.

**Comment by sercanov** [comment]
Rather than yeoman we can also use [plopjs](https://github.com/amwmedia/plop) and [gluegun](https://github.com/infinitered/gluegun), they all can do the job but plop.js seems more lightweight and easy to implement. 

I'll come up with experimental branch in few days.

**Comment by sercanov** [comment]
It's here https://github.com/sercanov/pepperoni-app-kit/tree/feat/generator

What do you think ?

**Comment by krivachy** [comment]
Hey @sercanov, do you mind opening a PR? Would be easier to review and keep track of it. Based on my initial glance it looks nice. I think what we also want in the future is to be able to add extra "features" to a sample app, such as Auth0 (removed due to many people not liking it) and other optional things.

**Comment by sercanov** [comment]
Hey @krivachy sure, opening right away.

Hey actually there are a lot of to do and plenty of possibilities. It may evolve a whole selective generator in the end.
